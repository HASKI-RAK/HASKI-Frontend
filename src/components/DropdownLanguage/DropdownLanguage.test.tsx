import renderer from "react-test-renderer";
import "@testing-library/jest-dom";
import { DropdownLanguage } from "@components";
import {fireEvent, getByRole, getAllByRole, render, act} from "@testing-library/react";



// tests for mui can be found https://github.com/mui/material-ui/blob/master/packages/mui-material/src

describe("Test the change-language dropdown component", () => {

    test("dropdown can be set to german", () => {

        const {getAllByRole,getByRole} = render(<DropdownLanguage/>);

        fireEvent.mouseDown(getByRole('button'));
        act(() => {
            getAllByRole('option')[0].click();
        })

        expect(getByRole('button')).toHaveTextContent(/Deutsch/i);
    });
    
    test("dropdown can be set to english", () => {
        
        
        const {getAllByRole,getByRole} = render(<DropdownLanguage/>);
        
        fireEvent.mouseDown(getByRole('button'));
        act(() => {
            getAllByRole('option')[1].click();
        })

        expect(getByRole('button')).toHaveTextContent(/English/i);
    });
    
    test("renders correctly", () => {
        const tree = renderer
            .create(
                <DropdownLanguage/>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});