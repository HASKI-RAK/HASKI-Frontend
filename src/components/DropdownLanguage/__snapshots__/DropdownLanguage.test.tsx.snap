// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the change-language dropdown component renders correctly 1`] = `
<div>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary LanguageDropdown css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
    onClick={[Function]}
  >
    <div
      aria-expanded="false"
      aria-haspopup="listbox"
      className="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="button"
      tabIndex={0}
    >
      English
    </div>
    <input
      aria-hidden={true}
      className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
      onAnimationStart={[Function]}
      onChange={[Function]}
      tabIndex={-1}
      value="en"
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
      data-testid="ArrowDropDownIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-ihdtdm"
      >
        <span
          className="notranslate"
        >
          â€‹
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;
